python/ml4ir/applications/ranking/model/metrics/helpers/metrics_helper.py

```diff
@@ -62,8 +137,15 @@ def get_grouped_stats(
         computed from the old and new ranks and aux labels generated
         by the model
     """
-    # Filter unclicked queries
-    df_clicked = df[df[label_col] == 1.0]
+    # adding "top_graded_relevance" to be the artificial click such that the record with the highest graded
+    # relevance is considered to be clicked nad will be used in MRR computation
+    artificial_click_col = "top_graded_relevance"
+    df = add_top_graded_relevance_record_column(df, label_col, artificial_click_col)
+    if np.array([Metric.NDCG in m for m in power_analysis_metrics]).any():
+        df = compute_ndcg(df, label_col, pred_col="ranking_score", new_col=RankingConstants.NEW_NDCG)
+        df = compute_ndcg(df, label_col, pred_col="s", new_col=RankingConstants.OLD_NDCG)
+
+    df_clicked = df[df[artificial_click_col] == 1.0]
```

Comment:
@mohazahran Update existing tests to capture these new metrics

Reply:
done